src/leiningen/auto.clj

```diff
@@ -36,51 +38,64 @@
   (binding [main/*exit-process?* false]
     (main/resolve-and-apply project (cons task args))))
 
-(defn add-ending-separator [^String path]
-  (if (.endsWith path File/separator)
-    path
-    (str path File/separator)))
-
-(defn remove-prefix [^String s ^String prefix]
-  (if (.startsWith s prefix)
-    (subs s (.length prefix))
-    s))
-
-(defn show-modified [project files]
-  (let [root  (add-ending-separator (:root project))
-        paths (map #(remove-prefix (str %) root) files)]
-    (str/join ", " paths)))
-
 (def default-config
   {:file-pattern #"\.(clj|cljs|cljx|cljc)$"
-   :wait-time    50
    :log-color    :magenta})
 
 (defn default-paths [project]
   (concat (:source-paths project)
           (:java-source-paths project)
           (:test-paths project)))
 
+(def watched-dirs (atom {}))
+
+(def watch-service
+  (delay (.newWatchService (FileSystems/getDefault))))
+
+(defn path-from-event [watchkey evt]
+  (str (get @watched-dirs watchkey) "/" (-> evt .context .toString)))
+
+(defn watch-dir! [dir]
+  (let [key (.register (.toPath dir) @watch-service
+              (into-array [StandardWatchEventKinds/ENTRY_CREATE
+                          StandardWatchEventKinds/ENTRY_MODIFY
+                          StandardWatchEventKinds/ENTRY_DELETE])
+              (into-array [SensitivityWatchEventModifier/HIGH]))]
```

Comment:
Does it need this fourth argument?

Reply:
It does work without, but responds much faster with it